invoker/core/src/main/java/com/google/cloud/functions/invoker/http/TimeoutFilter.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.functions.invoker.http;
+
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+public class TimeoutFilter implements Filter {
+  private static final int DEFAULT_TIMEOUT_SECONDS = 30; // Default timeout in seconds
+  private final int timeoutMs;
+
+  public TimeoutFilter() {
```

Comment:
Should this be a private default constructor?  30 seconds seems like a kind of magical default for functions framework to enforce.

Reply:
I'll just remove the default & stick to normal behavior :P

Reply:
The default is never used anyway, and timeout seconds is only set when the env var is provided